src/parser.js

```diff
@@ -1084,26 +1143,29 @@ function insertVariableDeclarations(ast) {
     }
     const needle = path.value.left.name;
     if (!path.scope.lookup(needle)) {
-      if (path.parent && path.parent.value.type === 'ExpressionStatement') {
-        if (isArrayAssignmentWithThisReference(path)) {
-          insertArrayAssignmentVarDeclarations(path);
-        } else {
-          jsc(path).replaceWith(_path =>
-            b.variableDeclaration(
-              'var',
-              [b.variableDeclarator(
-                _path.value.left,
-                _path.value.right
-              )]
-            )
-          );
-          path.scope.scan(true);
-        }
+      const blockNode = findNodeParent(path, node => get(node, 'value.type') === 'BlockStatement');
+      if (path.parent && path.parent.value.type === 'ExpressionStatement' && isArrayAssignmentWithThisReference(path)) {
+        insertArrayAssignmentVarDeclarations(path);
+      } else if (path.parent &&
+          get(path, 'parent.parent.value.type') !== 'SwitchCase' &&
```

Comment:
Merge conflict here ([mainline code](https://github.com/juliankrispel/decaf/blob/master/src/parser.js#L975)).  I made the `!== SwitchCase` logic come after our `isArrayAssignmentWithThisReference` logic.

I'll submit the `isArrayAssignmentWithThisReference` fix up as a PR to mainline.